dashboard/pages/primary_screening/callbacks.py

```diff
@@ -0,0 +1,106 @@
+from dash import html, no_update, callback, Output, Input, State
+
+from ..error import throwable
+from ...data.bmg_plate import parse_bmg_files_from_iostring
+from ...data.file_preprocessing.echo_files_parser import EchoFilesParser
+import io
+import base64
+import functools
+import uuid
+
+
+@throwable(1)
+def upload_bmg_data(contents, names, last_modified, stored_uuid, file_storage):
+    if contents is None:
+        return no_update
+
+    if not stored_uuid:
+        stored_uuid = str(uuid.uuid4())
+
+    bmg_files = []
+
+    for content, filename in zip(contents, names):
+        name, extension = filename.split(".")
+        if extension == "txt":
+            _, content_string = content.split(",")
+            decoded = base64.b64decode(content_string)
+            bmg_files.append((filename, io.StringIO(decoded.decode("utf-8"))))
+
+    if len(bmg_files):
+        bmg_df, val = parse_bmg_files_from_iostring(tuple(bmg_files))
+        serialized_processed_df = bmg_df.reset_index().to_parquet()
+        file_storage.save_file(f"{stored_uuid}_bmg_df.pq", serialized_processed_df)
+
+    return (
+        html.Div(
+            [
+                html.H5("Loaded files"),
+                html.Hr(),
+                html.Ul(children=[html.Li(i) for i in names]),
+                html.Hr(),
+            ]
+        ),
+        stored_uuid,
+    )
+
+
+@throwable(1)
+def upload_echo_data(contents, names, last_modified, stored_uuid, file_storage):
+    if contents is None:
+        return no_update
+
+    if not stored_uuid:
+        stored_uuid = str(uuid.uuid4())
```

Comment:
It does not make sense to create a random UUID in here if you will not save it in the Client's memory

It should work as previously - the first callback generates the UUID for the user and stores in on his side in browsers' `localStorage`

Then all subsequent callbacks that deal with files, retrive the stored UUID to determine which file refers to which user.